tests/behat/behat_repository_ocis.php

```diff
@@ -0,0 +1,86 @@
+<?php
+// This file is part of the ocis repository for Moodle - http://moodle.org/
+//
+// The ocis repository for Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The ocis repository for Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with the ocis repository for Moodle.  If not, see <http://www.gnu.org/licenses/>.
+//
+// ownCloud Infinite Scale for moodle Repository is built with the contributions of:
+// - Staatsinstitut für Schulqualität und Bildungsforschung
+// - JankariTech
+// - ownCloud - a Kiteworks company.
+
+/**
+ * Steps definitions related to repository_ocis.
+ *
+ * @package    repository_ocis
+ * @copyright  2017 Project seminar (Learnweb, University of Münster)
+ * @copyright  2023 ownCloud GmbH
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+// NOTE: no MOODLE_INTERNAL test here, this file may be required by behat before including /config.php.
+
+require_once(__DIR__ . '/../../../../lib/behat/core_behat_file_helper.php');
+
+require_once(__DIR__ . '/../../vendor/autoload.php');
+
+use Sabre\DAV\Client,
+    Behat\Behat\Hook\Scope\BeforeScenarioScope,
+    Behat\Behat\Hook\Scope\AfterScenarioScope;
+
+/**
+ * Steps definitions related to repository_ocis.
+ */
+class behat_repository_ocis extends behat_base {
+    private Client $client;
+    public function __construct() {
+        $settings = [
+            'baseUri' => getenv('MOODLE_OCIS_URL'),
+            'userName' => 'admin',
+            'password' => 'admin',
+        ];
+        $this->client = new Client($settings);
+    }
+
+    /**
+     * @BeforeScenario
+     * @param BeforeScenarioScope $scope The Behat Scope
+     */
+    public function set_default_editor_flag(BeforeScenarioScope $scope): void {
```

Comment:
I don't understand the function name. This function seems to upload a file, but the name says something unrelated